miio/gateway.py

```diff
@@ -647,6 +647,11 @@ def device_type(self):
         """Return the device type name."""
         return self.type.name
 
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"{self.device_type}-{self.sid}"
```

Comment:
So, I was thinking the the "name" should be something user-friendly, like a real description of the device (and maybe its component number). Something like what is shown at https://www.zigbee2mqtt.io/information/supported_devices.html#xiaomi

Reply:
I am not sure how we would then implement that:
Don't you think "double key wireless wall switch" and "temperature, humidity and pressure sensor" are a bit long for a name, espacially if we also include the sid...

Reply:
I could just make a `_name` variable for each subdevice that is implemented and we could then discuss the individual names, does that sound good?

furthermore I see the models are here also specified.
Schould I rename the current _model variable to _zigbee_id and add those models
So for the magnet it would be:
_model = "MCCGQ11LM"
_zigbee_id = "lumi.sensor_magnet.aq2" (see https://zigbee.blakadder.com/Xiaomi_MCCGQ11LM.html)

Reply:
That's what zigbee2mqtt is using, so I don't think it's that bad:

![image](https://user-images.githubusercontent.com/3705853/84447763-3d582b80-ac49-11ea-9231-012568925bd8.png)

There is no unique id here though, but I'm not sure that is something that belongs to the user interface anyway.

Reply:
Separating the zigbee_id from the name/model sounds good to me. I'm not sure how much information the "model number" itself does, except that it makes it easier to google for it.

For the sensors in homeassistant we want to have some sort of human-readable string, in my opinion at least. The rest can easily be adapted/added later on, that's why I kind of like the "nice descriptive name (model number)" format zigbee2mqtt uses.

Reply:
I will start on this tommorow.
However having the sid in the name is very usefull if you have multiple of the same devices.
In my case I have 3 temperature sensors and I need to figure out which one is which. I can do that using the sid because the sid can be found in the MiHome app where I originally linked the devices and gave them proper names so I know which one is which.

Besides these are only the default names, 95% of the users will change the name anyway once they set up the integration.

Reply:
Ah, indeed, I can see definitely how that can be useful. Another idea, could the sid be exposed as a separate attribute in homeassistant instead of embedding it into the name?

I don't really care that much as I'm not personally using this, but I think having "Aqara Temperature sensor ... (sid)" is friendlier than "AqaraHT-123123123213". I agree, that the sid should somehow be exposed especially as it can be useful with the app as you mentioned.

Reply:
that would be ideal but currently the only info that can be shown is:
manfacturer
model
fw_ver
connected_via_device

I would defenitly like to expose more attributes but that is currently not possible.
What I would like:
serial_number (I would use sid for xiaomi_miio)
protocol_version (not for xiaomi_miio)
zigbee_model
mac_adress (not for xiaomi_miio)
ip_adress

All of those would be nice to have under device_info...
But that is not going to happen any time soon